CRTT0Tagging - icaruscode v10

francescopoppi opened the pull request:
This is the big Pull Request which will introduce CRTT0Tagging into Stage1. CRTT0Tagging will be used at the moment for introducing CRT Tagged Tracks into the calibration Ntuples. The core of the software was partly already reviewd by @PetrilloAtWork , since I believe this is the definitive Pull Request I am requesting an additional check.
In this code legacy CRTTagging code has been moved to Legacy Folder.
I am also requesting a review from @SFBayLaser because I would like guidance on the stage1 fcl and stage1 definitions to modify, I have a proposal here, but should be double checked.
I also included here @jzennamo and @gputnam because this PR is needed for the Spring Production project.
This require a dependance on two new SBNObj Data Products, I will add a comment here with the PR as soon as it is created.

francescopoppi commented:
Ok, the SBNObj PR needed for this PR was already open by @gputnam and it is this one: https://github.com/SBNSoftware/sbnobj/pull/124

leoaliaga commented:
Hello, the required SBNObj PR has already been merged and is included in sbnobj v10_00_09. Can we proceed with reviewing this PR? Just to note: the new module CRTT0Tagging will be included when we merge [PR-774

SFBayLaser commented on fcl/reco/Definitions/stage1_icarus_defs.fcl:
```diff
@@ -65,9 +68,11 @@ icarus_stage1_producers:
 
  ## crt producer
   crttrack:          @local::standard_crttrackproducer
-  CRTT0Matching:     @local::standard_crtt0producer
-  CRTT0MatchingW:    @local::standard_crtt0producerW
-  CRTT0MatchingE:    @local::standard_crtt0producerE
+  ### FPoppi: the following are deprecated (and the one above not mantained)
```
My suggestion is if the code no longer exists or is supported that it be removed from  here to avoid future confusion.

francescopoppi replied:
Ok Tracy, I am going to remove it!

francescopoppi commented:
https://github.com/SBNSoftware/icaruscode/blob/cd7ba42cdfbff486edfbf07f11d2974061d9ac07/fcl/caf/cafmaker_defs.fcl#L91
There is one thing I did not see before. Apparently 2 years ago @gputnam added the old crt t0 producer in the cafs definitions. Since this is now done at stage1, can I remove this?

francescopoppi commented:
@SFBayLaser I removed deprecated producers from the Stage1 definitions and I also removed duplicated CRTT0 production at caf levels according to the modifications that @gputnam did on the release/Mar25Production branch.

leoaliaga commented:
@SFBayLaser, @gputnam, @PetrilloAtWork, @jzennamo: Could you please take a look at this PR. is it ready or needs further revisions? Thanks!

leoaliaga commented:
trigger build LArSoft/lar*@LARSOFT_SUITE_v10_06_00 SBNSoftware/sbncode@v10_06_00 SBNSoftware/sbnalg@v10_06_00 SBNSoftware/sbnobj@v10_01_00 SBNSoftware/icarus_data@v09_93_05

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/26270&builds=)

FNALbuild commented:
:x: CI build for ICARUS Failed at phase build ICARUS on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/12046&builds=)

:rotating_light: For more details about the failed phase, check the [build ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/12046&platform=Linux%20slf7&phase=build&buildtype=slf7%20c14:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/26271&builds=)

FNALbuild commented:
:x: CI build for ICARUS Failed at phase build ICARUS on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/12045&builds=)

:rotating_light: For more details about the failed phase, check the [build ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/12045&platform=Linux%20slf7&phase=build&buildtype=slf7%20e26:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/26270&builds=)

SFBayLaser reviewed (approved):
Good to go for me

PetrilloAtWork commented on icaruscode/CRT/CRTLegacyCode/CRTTzeroProducer_module.cc:
The next one will become `CRTTnaughtProducer`?

francescopoppi commented:
I included some changes in fcl parameters to do 2 things:
a) give more freedom of selection to the analyzers
b) force common cuts for calibration tuple calorimetry

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/26375&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/26376&builds=)

FNALbuild commented:
:x: CI build for ICARUS Failed at phase build ICARUS on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/12135&builds=)

:rotating_light: For more details about the failed phase, check the [build ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/12135&platform=Linux%20slf7&phase=build&buildtype=slf7%20c14:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/26376&builds=)

FNALbuild commented:
:x: CI build for ICARUS Failed at phase ci_tests ICARUS on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/12134&builds=)

:rotating_light: For more details about the failed phase, check the [ci_tests ICARUS phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:ICARUS/build_detail/phase_details?build_id=icarus_ci/12134&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20e26:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/26375&builds=)
